src/models/hooks/activityReport.js

```diff
@@ -797,43 +799,45 @@ const autoPopulateUtilizer = async (sequelize, instance, options) => {
           activityReportId: instance.id,
         },
         raw: true, // Return raw data instead of Sequelize instances
+        transaction: options.transaction,
       }), // End of the second query
     ]);
 
     const users = [
       ...collaborators, // Spread the elements of the 'collaborators' array into a new array
       { userId: instance.userId }, // Add an object with a 'userId' property to the new array
     ].filter(({ userId }) => userId);
-    await Promise.all([
-      ...users
-      // Use flatMap to iterate over each element in the new array asynchronously
-        .flatMap(async ({ userId }) => goals
-          // Use map to iterate over each element in the 'goals' array asynchronously
-          // Call the 'findOrCreateCollaborator' function with the following arguments
-          .map(async ({ goalId }) => findOrCreateCollaborator(
-            'goal',
-            sequelize, // The 'sequelize' variable
-            options.transaction, // The 'options' variable
-            goalId, // The 'goalId' from the current iteration of the 'goals' array
-            userId, // The 'userId' from the current iteration of the new array
-            GOAL_COLLABORATORS.UTILIZER, // The 'GOAL_COLLABORATORS.UTILIZER' constant
-            { activityReportIds: [instance.id] },
-          ))),
-      ...users
-      // Use flatMap to iterate over each element in the new array asynchronously
-        .flatMap(async ({ userId }) => objectives
-          // Use map to iterate over each element in the 'objectives' array asynchronously
-          // Call the 'findOrCreateCollaborator' function with the following arguments
-          .map(async ({ objectiveId }) => findOrCreateCollaborator(
-            'objective',
-            sequelize, // The 'sequelize' variable
-            options.transaction, // The 'options' variable
-            objectiveId, // The 'objectiveId' from the current iteration of the 'objectives' array
-            userId, // The 'userId' from the current iteration of the new array
-            OBJECTIVE_COLLABORATORS.UTILIZER, // The 'OBJECTIVE_COLLABORATORS.UTILIZER' constant
-            { activityReportIds: [instance.id] },
-          ))),
+
```

Comment:
How is this achieve something different? It looks like it would have the same effect?

Reply:
flatMap is not waiting for the result of async findOrCreateCollaborator